turtle.js

```diff
@@ -429,6 +429,38 @@
       }
     }},
 
+    ellipse: {value: function(width, height) {
+      var self = this; 
+      var x = self.x ;
+      var y = self.y ;
+      x = x || 0;
+      y = y || 0;
+
+      if (width <= 0 && height <= 0) {
+        return;
+      }
+
+      if (width === height) {
+        this.canvas_ctx.beginPath();
```

Comment:
Look at the ARC implementation to see how `filling` needs to be tested - there are two different modes (drawing and filling).